client/src/pages/Notification/NotificationTable/DiscordTable.tsx

```diff
@@ -0,0 +1,156 @@
+import Table, { TableColumn } from 'components/Table/Table';
+import NotificationButtons from 'pages/Notification/NotificationButtons/NotificationButtons';
+import HiddenWebhook from 'pages/Notification/HiddenWebhook/HiddenWebhook';
+import NotificationNameLink from 'pages/Notification/NotificationNameLink/NotificationNameLink';
+import SectionComponent from 'components/SectionComponent/SectionComponent';
+import discordIcon from 'assets/discord-icon.png';
+import { NotificationTableRowProps } from './NotificationTable';
+import { useEffect, useState } from 'react';
+import {
+  DiscordNotificationChannel,
+  ManagmentServiceApiInstance,
+} from 'api/managment-service';
+import LoadingTable from './LoadingTable';
+import NewDiscordChannelPopup from 'pages/Notification/NewChannelPopup/NewDiscordChannelPopup';
+import { dateFromTimestampMs } from 'lib/date';
+import EditDiscordChannelPopup from 'pages/Notification/EditChannelPopup/EditDiscordChannelPopup';
+import './NotificationTable.scss';
+
+interface DiscordTableRowProps extends NotificationTableRowProps {
+  webhookUrl: string;
+}
+
+const getDiscordChannelTableRow = ({
+  id,
+  receiverName,
+  updatedAt,
+  createdAt,
+  webhookUrl,
+}: DiscordNotificationChannel): DiscordTableRowProps => ({
+  name: receiverName,
+  updatedAt: dateFromTimestampMs(updatedAt),
+  createdAt: dateFromTimestampMs(createdAt),
+  webhookUrl: webhookUrl,
+  id,
+});
+const DISCORD_WEBHOOK_FORMAT =
+  'https://discord.com/api/webhooks/<webhookId>/<authToken>';
+
+export const discordWebhookValidation = (webhook: string) => {
+  const discordWebhookRegex =
+    /^https:\/\/discord\.com\/api\/webhooks\/\d+\/[a-zA-Z0-9_-]+$/;
```

Comment:
consider extracting this into separate file

Reply:
Do you mind elaborating on this idea?

Reply:
move this pattern to new file just for patterns. I would be much easier in future to change that If needed